LEGO-9977 - input: Вынести реализацию onchange на таймерах на уровень desktop

Lavr1k opened the pull request:
1) Вынес все связанное с десктопами на уровень к desktop
2) WP 7-8 : Увеличил область крестика путем уменшения отступа инпута
3) Увеличил вес селектора чтобы в msie10 левый отступ был равен нулю и input__found не налазил на текст запроса

Lavr1k committed:
[WinPhone 7-8] не кликабельный крестик в инпуте

Увеличил область крестика путем уменшения отступа инпута

Lavr1k committed:
input: вынести реализацию onchange на таймерах на уровень desktop

Вынес все связанное с десктопами на уровень к desktop

gela-d commented on desktop.blocks/input/__box/input__box.ie8.css:
```diff
@@ -0,0 +1,10 @@
+.input__box
```
Это ж десктоп, а в задаче идет вроде речь о wp?

Lavr1k replied:
Это тоже для ие8)

gela-d commented on desktop.blocks/input/__clear/input__clear.ie8.css:
```diff
@@ -0,0 +1,9 @@
+.input__clear
+{
```
это все точно не сюда :)

Lavr1k replied:
Это для ие8 я зачем-то крестик сделал)

gela-d replied:
=))) Саша, не надо крестик, тем более он svg, а ие8 не поддерживает его.
Дойти до Вовы, там надо все переделать)

gela-d commented:
Вова, глянь, пожалуйста

Lavr1k committed:
input: убрать padding-left у элемента control

Увеличил вес селектора, чтобы в msie10 левый отступ был равен нулю и input__found не налазил на текст запроса

tadatuta commented on common.blocks/input/__clear/input__clear.js:
```diff
@@ -45,6 +43,21 @@ BEM.DOM.decl('input', /** @lends Block.prototype */ {
+
+        var _this = this;
+
+        _this.bindTo('input', function() {
```
это что-то ОЧЕНЬ странное o_0
что такое событие `input`?

apostololeg replied:
[GlobalEventHandlers.oninput](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers.oninput?redirectlocale=en-US&redirectslug=Web%2FAPI%2Fwindow.oninput) ;)

Lavr1k replied:
Ох, да )
Какието проблемы с подпиской на `change`

tadatuta commented:
Саш, дойди до меня, тут много странного...

gela-d commented:
Дим, судя по задаче, там функциональность, которая не нужна на тачах.

dfilatov commented:
Напишите кто-нибудь, чем именно текущая реализация на таймерах мешает правильной работе в тачах. В таске только общие слова.

deepsweet commented:
необъяснимые проблемы с вводом текста, проблемы с placeholder'ами, проблемы с вставкой текста по longtap+paste, проблемы с получением val() после вставки, и ещё бог знает что.
да и банально каждый таймер – это удар по аккумулятору.
инпут сейчас – основная головная боль с миллионом тасков, при том что в ромочке, где распилено изначально и чуть ли не переписано с нуля на нативные события, таких проблем нет.

deepsweet commented:
мы хотим не столько "вынести ненужное", сколько решить проблемы.

dfilatov commented:
Ясно. А в тачевых браузерах есть автозаполнение?

tadatuta commented:
перекрестился
